specs/bep-v1.rst

```diff
@@ -844,6 +862,140 @@ XDR
+    /                                                               /
+    \                  Name (length + padded data)                  \
+    /                                                               /
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /                                                               /
+    \                      Version Structure                        \
+    /                                                               /
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                    Number of Block Indexes                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /                                                               /
+    |                    Block Indexes (n items)                    |
+    /                                                               /
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+Fields (DownloadProgress Message)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+**Folder** represents the ID of the folder for which the update is being
+provided.
+
+The **Flags** field is reserved for future use and MUST currently be set to
+zero. The **Options** field contains a list of options that apply to the update.
+
+Fields (FileDownloadProgressUpdate Structure)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The **Update Type** field is made up of the following single bit flags:
+::
+
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                           Reserved                          |F|
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+:Bit 31 ("F", Forget): is set to notify that the file that was previously
+    advertised is no longer available (as a temporary file).
+
+The **Name** field defines the file name from the global index for which this
+update is being sent.
+
+The **Version** structure defines the version of the file for which this update
+is being sent.
+
+**Block Indexes** is a list of positive integers, where each integer represents
+the index of the block in the FileInfo structure Blocks array that has become
+available for download.
+For example an integer with with value 3 represents that the data defined in the
+third BlockInfo structure of the FileInfo structure of that file is now available.
+Please note that matching should be done on **Name** AND **Version**.
+Furthermore, each update received is incremental, for example the initial 
+FileDownloadProgressUpdate structure might contain indexes 0, 1, 2, an update 
+5 seconds later might contain indexes 3, 4, 5 which should be appended to the 
+original list, which implies that blocks 0-6 are currently available.
+
+Blocks are downloaded in random order, therefore block indexes MAY NOT grow in
```

Comment:
Skip the part about blocks being downloaded in random order as that's implementation dependent, but the conclusion about the indexes not growing in any defined order should be in here as it's true.